### PR TITLE
Make 'default_frame_layout' a LayoutAlgorithm setting

### LINKED ISSUE #24: Commands which accept layouts should accept names
Currently commands which take a layout expect a hard to remember integer - we should be able to use layout names instead (or both, to not break backwards compatibility)

### PR DESCRIPTION
**Pull request by t-wissmann**
This makes the setting 'default_frame_layout' have the type
LayoutAlgorithm, so one can now write human-readable names to it rather
than indices. For backwards-compatibility, indices are still accepted
for parsing, but they are neither printed nor completed. The indices are
also removed from the man page to avoid the confusion reported in #24.

On the technical side, the ValueList of a 'Finite<>' type now also holds
a flag that tells whether parsing should interpret integers as indices
in the list of values. This flag is only activated for LayoutAlgorithm
and is 'false' in all other cases.

The test cases are extended accordingly.

This fixes #24.

### CODE REVIEW

**Comment by blarz** [comment]
Yes I love it, this always confused me because I had to look up the number of the max_layout ;)

File: tests/test_settings.py [hunk 1]
```diff
@@ -72,12 +72,61 @@ def test_cycle_value_loop(hlwm):
```
Comment (blarz) [inline tests/test_settings.py:72]: Can you add a test for an invalid layout algorithm string? E.g. 
```set default_frame_layout foobar```

Comment (blarz) [reply tests/test_settings.py:72]: Ah there already is one with ```test_default_frame_layout_value_negative```.
Maybe we can rename the test? It doesn't fit that nicely anymore IMHO?

Comment (t-wissmann) [reply tests/test_settings.py:72]: I've renamed it. (And I'm testing for `-1` and `foobar`.

Comment (blarz) [reply tests/test_settings.py:72]: Thanks!

**Review by blarz — APPROVED** [review]
Cool!

**Pull request merged**